### PR TITLE
runtime: support cgroup v2 metrics marshal guest metrics

### PR DESCRIPTION
**Pull request by yaoyinnan**
Support to use cgroup v2 metrics marshal guest metrics.

Fixes: #5998

Signed-off-by: yaoyinnan <yaoyinnan@foxmail.com>

### CODE REVIEW

**Comment by liubin** [comment]
CI error:

> Error: metrics_test.go:57:13: cannot initialize 1 variables with 2 values (typecheck)
	metrics := statsToMetrics(&resp)

**Comment by yaoyinnan** [comment]
> CI error:
> 
> > Error: metrics_test.go:57:13: cannot initialize 1 variables with 2 values (typecheck)
> > metrics := statsToMetrics(&resp)

I split `statsToMetrics()` and changed it to judge whether it is cgroup v1 or v2 in `marshalMetrics()`. The unit test `TestStatNetworkMetric()` is only tested when the cgroup is v1.
This modification is due to two problems: the first problem is that the stats returned by the agent do not contain `networkStats`, and there is no other way to obtain networkStats, refer to https://github.com/kata-containers/kata-containers /blob/main/src/agent/rustjail/src/container.rs#L856. The second problem is that the metrics of cgroup v2 do not include networkStats.

**Review by liubin — APPROVED** [review]

**Comment by bergwolf** [comment]
Please rebase this, thanks!

**Comment by yaoyinnan** [comment]
> Please rebase this, thanks!

Rebased, thanks.

File: src/runtime/pkg/containerd-shim-v2/metrics.go [hunk 1]
```diff
@@ -175,6 +258,28 @@ func copyBlkio(s []vc.BlkioStatEntry) []*cgroupsv1.BlkIOEntry {
 	return ret
 }
 
+func copyBlkioV2(s []vc.BlkioStatEntry) []*cgroupsv2.IOEntry {
+	var ret []*cgroupsv2.IOEntry
+	var item *cgroupsv2.IOEntry
```
Comment (luodw) [inline src/runtime/pkg/containerd-shim-v2/metrics.go:263]: here item is null pointer?

Comment (yaoyinnan) [reply src/runtime/pkg/containerd-shim-v2/metrics.go]: fixed

**Review by GabyCT — APPROVED** [review]

**Pull request merged**